htmresearch/algorithms/column_pooler.py

```diff
@@ -174,16 +192,127 @@ def compute(self, feedforwardInput=(), lateralInputs=(),
 
     @param  learn (bool)
             If True, we are learning a new object
-    """
 
+    @param predictedInput (sequence)
+           Sorted indices of active feedforward input bits
+    """
     if feedforwardGrowthCandidates is None:
       feedforwardGrowthCandidates = feedforwardInput
 
-    if learn:
+    if not learn:
+      self._computeInferenceMode(feedforwardInput, lateralInputs)
+
+    elif not self.onlineLearning:
       self._computeLearningMode(feedforwardInput, lateralInputs,
                                 feedforwardGrowthCandidates)
     else:
-      self._computeInferenceMode(feedforwardInput, lateralInputs)
+      if (predictedInput is not None and
+          len(predictedInput) > self.predictedInhibitionThreshold):
```

Comment:
That's correct, it could never learn "A", unless you were to replay the sequence with the "APQRS" SDR still active.  Fundamentally, I think this is an intractable problem without either backtracking or a supervisory signal, and it's outside of the scope of this to have either.